Remove ShellJS copy and use the NPM version

timvandermeij opened the pull request:
There is no need to have a copy of ShellJS in the repository as it is also available on NPM. The NPM version is also much newer. This way we do not have to update this anymore and let NPM do that automatically.

timvandermeij committed:
Remove ShellJS copy and use the NPM version

There is no need to have a copy of ShellJS in the repository as it is also available on NPM. The NPM version is also much newer. This way we do not have to update this anymore and let NPM do that automatically.

timvandermeij commented:
/botio preview

Snuffleupagus commented on make.js:
```diff
@@ -20,7 +20,7 @@
 'use strict';
 
-require('./external/shelljs/make');
+require('shelljs/make');
```
I'm wondering if it would be a good idea to enclose this in a `try... catch` block, e.g. something like this:
```javascript
try {
  require('shelljs/make');
} catch (ex) {
  console.log('Run "npm install"'); // Probably needs a better message
  return;
}
```
When I tested the patch the first time, running any of the `make` commands just threw an error, and it took me a couple of seconds to realise that I needed to run `npm install`.
For less experienced users, it's not unlikely that they *could* get stuck on this, so a message telling them what to do seems like a good idea to me.

*Edit:* A somewhat fun fact is that if the above is implemented, I think that it would actually make PR #5908 obsolete :-)

timvandermeij replied:
Done. Great catch, thanks for noticing this!

timvandermeij committed:
Remove unreachable jshint installation code

The previous commit implements a check for ShellJS and otherwise prompts the user to run "npm install". New clones of the codebase will need "npm install" for ShellJS and therefore automatically install jshint. Existing clones of the codebase will also need "npm install" again since ShellJS needs to be installed using NPM as it is not in the "external" folder anymore. Since everyone will get this prompt and install everything automatically, we will never reach this code path anymore.

This patch makes #5908 obsolete and reduces code complexity for the lint target. Thanks to @Snuffleupagus for noticing this!

timvandermeij commented:
/botio preview

Snuffleupagus commented:
This does look perfectly fine to me, and it'd be great to be able to get rid of all this code from the repo!
However before merging this, I want to check with Brendan/Yury on IRC, to make sure that there are not any weird dependencies in edge cases (I'm mostly concerned about the bots).

/botio test

Snuffleupagus commented:
Thank you!

yurydelendik commented:
@timvandermeij we probably need to fix version number to be exact to avoid changes similar to https://github.com/arturadib/shelljs/pull/207
